flink-runtime/src/test/java/org/apache/flink/runtime/executiongraph/ExecutionTest.java

```diff
@@ -361,11 +168,7 @@ public void testTaskRestoreStateIsNulledAfterDeployment() throws Exception {
 		assertThat(execution.getTaskRestore(), is(notNullValue()));
 
 		// schedule the execution vertex and wait for its deployment
-		executionVertex.scheduleForExecution(
-			executionGraph.getSlotProviderStrategy(),
-			LocationPreferenceConstraint.ANY,
-			Collections.emptySet())
-			.get();
+		scheduler.startScheduling();
```

Comment:
`scheduler.startScheduling` does not look to wait for anything, would it be easier to call `Execution::deploy`?

Reply:
The slot request will be immediately fulfilled and trigger the deployment of the Execution.
This is also the previous assumption in this test.
I'd like to avoid directly call `Execution::deploy` because this is also one of the actions that we'd like to factor out from Execution graph components, although we do not rework it right now.